### PR TITLE
fix(carousel): centralize the slide item content

### PR DESCRIPTION
**Pull request by MayaraRMA**
## Infos

[CATÁLOGO - MOBILE - Responsividade carrossel ](https://juntossomosmais.monday.com/boards/3957397225/pulses/6100466635?userId=20950543)

#### What is being delivered?

- Centralize slide-item content

#### What impacts?

When the slide-item content is smaller than slide-item, the content will be in center.

#### Reversal plan

Describe which plan we should follow if this delivery has to be reversed.

#### Evidences
Screen Shot 2024-02-26 at 16 39 31

### CODE REVIEW

File: packages/core/src/components/carousel/carousel-item/carousel-item.scss
Comment (felipefialho) [inline packages/core/src/components/carousel/carousel-item/carousel-item.scss:1]: (block) we need start all scss files using `@import '~@atomium/scss-utils/index';`

File: packages/core/src/components/carousel/stories/style.css [hunk 1]
```diff
@@ -24,3 +24,12 @@
...
 }
+
+.item-smaller {
+  align-items: center;
+  background: coral;
+  display: flex;
+  height: 400px;
+  justify-content: center;
+  width: 250px;
+}
```
Comment (felipefialho) [inline packages/core/src/components/carousel/stories/style.css:35]: (question) do you want to simulate small screens in carousel? in theory we could just use the responsive tool in storybook

image

**Review by felipefialho — CHANGES_REQUESTED** [review]

File: packages/core/src/components/carousel/carousel-item/carousel-item.scss
Comment (MayaraRMA) [reply packages/core/src/components/carousel/carousel-item/carousel-item.scss:1]: sorry, I didn't see this pattern

File: packages/core/src/components/carousel/stories/style.css (hunk 1 above)
Comment (MayaraRMA) [reply packages/core/src/components/carousel/stories/style.css:35]: No. The idea is showing when the content is smaller than slider-item. I created Center Item Content 
Screen Shot 2024-02-22 at 15 24 56

File: packages/core/src/components/carousel/carousel-item/carousel-item.scss
Comment (MayaraRMA) [reply packages/core/src/components/carousel/carousel-item/carousel-item.scss:1]: Done [538f975](https://github.com/juntossomosmais/atomium/pull/385/commits/538f9753d69d0f04839ace922cedfbd68bf4d6f7)

**Review by felipefialho — APPROVED** [review]

File: packages/core/src/components/chip/chip.scss [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@import '~@atomium/scss-utils/index';
```
Comment (mauriciomutte) [inline packages/core/src/components/chip/chip.scss:1]: question: why do we need this sccs utility import?

**Review by mauriciomutte — APPROVED** [review]

File: packages/core/src/components/chip/chip.scss (hunk 2 above)
Comment (MayaraRMA) [reply packages/core/src/components/chip/chip.scss:1]: https://github.com/juntossomosmais/atomium/pull/385#discussion_r1499701612

**Review by felipefialho — APPROVED** [review]

**Comment by MayaraRMA** [comment]
com base nessa [thread](https://juntossomosmais.slack.com/archives/C0117K4MYSW/p1708612480098419) mudei a abordagem do que tinha sido feito e ficou mais de acordo com a biblioteca do swiper

**Review by felipefialho — DISMISSED** [review]

File: packages/core/src/components/carousel/carousel.tsx [hunk 3]
```diff
@@ -113,6 +110,13 @@ export class AtomCarousel {
     Object.assign(this.swiperEl, params)
   }
   render() {
+    const injectStyles = [
+      '.swiper-button-disabled { opacity: 0 !important}',
+      '.swiper-pagination-custom { display: flex; justify-content: center; column-gap: var(--spacing-base); } ',
+      `.swiper-button-next { width: var(--spacing-${this.navigationButtonSize}); height: var(--spacing-${this.navigationButtonSize}); }`,
+      `.swiper-button-prev { width: var(--spacing-${this.navigationButtonSize}); height: var(--spacing-${this.navigationButtonSize}); }`,
+    ]
```
Comment (felipefialho) [inline packages/core/src/components/carousel/carousel.tsx:117]: (question) why this approach?

Comment (MayaraRMA) [reply packages/core/src/components/carousel/carousel.tsx:117]: The swiper lib doesn't have a prop to change the navigation buttons size. 
Before `injectStyles` was a private prop. I dont know why, but I can't use a prop inside another prop, so this way I can manage the swiper classes make the buttons size dynamic.

**Review by felipefialho — APPROVED** [review]

**Review by mauriciomutte — APPROVED** [review]

**Pull request merged**